### DISCUSSION TITLE
JavaScript

### DISCUSSION THREAD

**Discussion by JuanPabloDiaz** [sphinx-doc/sphinx#11984] [Q&A]
I am looking for a document generator for my react projects and came across Sphinx. 

Does Sphinx support JavaScript framework? or is it only for python projects?

**Comment by Radhikacrypto** [comment]
Sphinx is primarily a documentation generator for Python projects, but it can be extended to support documentation for projects that involve other languages or frameworks, including those that use JavaScript

**Reply by JuanPabloDiaz** [reply]
Ohh okey,

I had some issues trying to implement Sphinx-js, and the documentation was not really clear. I got stuck in step 3. 

It seems like it's not beginner friendly 😕

**Reply by picnixz** [reply]
Unfortunately, we don't natively support JS. You can use `jsdoc` or `tsdoc` if your project in TypeScript. I'll close this conversation since you got an answer (maybe not satisfying).
